### DISCUSSION TITLE
Why does kobalte not use zag.js?

### DISCUSSION THREAD

**Discussion by samuelstroschein** [kobaltedev/kobalte#127] [General]
Hi @fabien-ml, 

I was wondering why you don't use ZagJS as primitives under the hood (we briefly tweeted). 

I just talked to David Di Biase. He shined some new light on why you don't use ZagJS. But, could you explain in your words why you don't use ZagJS?

PS We (inlang) are looking to adopt a component library. ZagJS looks promising but is a bit too low level. Kobalte seems good but too early and I fear that a lot of duplicate work is conducted/Kobalte could progress faster if ZagJS would be used.

**Comment by fabien-ml** [comment]
Hi,

- I don't like ZagJS API which is like React Aria (call function and spread props everywhere).
- I don't want to rely on a 3rd party library for the core of Kobalte.
- I want complete control over the API.
- I want to know how to build things on my own.

**Reply by samuelstroschein** [reply]
The reasoning is not performance related?

**Reply by fabien-ml** [reply]
No, I don't want to create a wrapper around Zag.

**Reply by samuelstroschein** [reply]
Besides not wanting to wrap Zag. Would it potentially speed up the development of Kobalte?

**Reply by fabien-ml** [reply]
I guess yes, because I will no longer need to worry about accessibility which is handled by Zag. 

Kobalte will be just a "component layer" on top of Zag, like [React Aria Components](https://github.com/adobe/react-spectrum/blob/aria-components-rfc/rfcs/2023-react-aria-components.md) is a layer on top of React Aria hooks.

**Reply by samuelstroschein** [reply]
Solid's bottleneck is the ecosystem which leads to migration discussions like https://github.com/inlang/inlang/discussions/420, wouldn't it make sense for Solid's and Kobalte's adoption to quickly build a component library that uses Zag to handle accessibility? 

You could ofc extend ZagJS, or build your own hooks in the future _after_ Kobalte serves the need of the majority of users.

**Reply by samuelstroschein** [reply]
I am trying to say that Solid needs to ramp up quickly to find its place in the JS world. I understand that full control is nice; understanding how everything works makes the engineering heart beat faster but ... users want/need to quickly built things. How Kobalte works under the hood is not a concern of users and makes people switch to React/Svelte/Vue instead :(

**Comment by fabien-ml** [comment]
I'm closing this since there is https://ark-ui.com which may fit your requirements.
